baseframe/forms/fields.py

```diff
@@ -725,3 +728,66 @@ def process_formdata(self, valuelist):
     def pre_validate(self, form):
         if self.data is _invalid_marker:
             raise ValueError(self.gettext('Not a valid choice'))
+
+
+class JsonField(wtforms.TextAreaField):
+    """
+    A field to accept JSON input, stored internally as a Python-native type.
+    By default, requires the JSON root object to be a dictionary/hash.
+
+    ::
+
+        class MyForm(forms.Form):
+            field = forms.JsonField(__("My Field"), default={})
+
+    :param str label: Field label
+    :param list validators: List of field validators, passed on to WTForms
+    :param bool require_dict: Require a dictionary as the data value (default `True`)
+    :param book use_decimal: Use decimals instead of floats (default `True`)
+    :param kwargs: Additional field arguments, passed on to WTForms
+    """
+    prettyprint_args = {'sort_keys': True, 'indent': 2}
+
+    def __init__(self, label='', validators=None, require_dict=True, use_decimal=True, **kwargs):
+        self.require_dict = require_dict
+        self.use_decimal = use_decimal
+        super(JsonField, self).__init__(label, validators, **kwargs)
+
+    def _value(self):
+        """
+        Render the internal Python value as a JSON string.
+        Specialcase `None` to return an empty string instead of a JSON ``null``.
+        """
+        if self.raw_data:
+            # If we've received data from a form, render it as is. This allows
+            # invalid JSON to be presented back to the user for correction.
+            return self.raw_data[0]
+        elif self.data is not None:
+            return json.dumps(self.data, use_decimal=self.use_decimal, ensure_ascii=False, **self.prettyprint_args)
+        return u''
```

Comment:
@jace missed a case here. When the db value is `{}` (for any model field with `default={}`), this returns `{}` to the html form instead of the form field default. You removed that part in your commit. Any reason why?